Bottom: 9a5b91cc56fc5798b88aabfa48e714335497962b
Top:    1d4a231b7810d43ef05dc76e524f10df3e2d4bf3
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:44:13 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2a19084..fc047f4 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1160,7 +1160,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
